src/Domain/BoundaryConditions/BoundaryCondition.hpp

```diff
@@ -0,0 +1,34 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <charm++.h>
+#include <memory>
+#include <pup.h>
+#include <string>
+
+#include "Parallel/CharmPupable.hpp"
+
+/// \ingroup ComputationalDomainGroup
+/// \brief %Domain support for applying boundary conditions
+namespace domain::boundary_conditions {
```

Comment:
I think we have no other namespace with an underscore, but have used lowercase names for short or abbreviated namespaces and CamelCase for composite names (e.g. `domain::CoordinateMaps`, `domain::FunctionsOfTime`, `grmhd::AnalyticData`, `ActionTesting`, `TimeSteppers`, ...). In particular for consistency with the other `domain` namespaces I would suggest to name this `domain::BoundaryConditions`.

Reply:
`pretty_type`. I prefer snake case namespaces. I'll change the domain one

Reply:
I would prefer consistency with the other `domain` namespaces over snake case, but if you lean the other way so be it 🤷‍♂️

Reply:
Fine